tests/store/crud/test_update.py

```diff
@@ -42,3 +45,23 @@ def test_update_user_is_archived(user_store: MockStore, user_email: str):
 
     # THEN user should be recorded as archived in the database
     assert archived_user
+
+
+def test_update_analysis_jobs_from_slurm_jobs(analysis_store: MockStore, squeue_stream_jobs: str):
+    """Test updating analysis jobs when given squeue results."""
+    # GIVEN an analysis and a squeue stream
+    analysis: Analysis = analysis_store.get_query(table=Analysis).first()
+    assert not analysis.failed_jobs
+
+    squeue_result: SqueueResult = get_squeue_result(squeue_response=squeue_stream_jobs)
+
+    # WHEN updating the analysis
+    analysis_store.update_analysis_jobs_from_slurm_jobs(
+        analysis=analysis, squeue_result=squeue_result
+    )
+    updated_analysis: Analysis = analysis_store.get_analysis(
+        case_id=analysis.family, started_at=analysis.started_at, status=TrailblazerStatus.PENDING
+    )
+
+    # THEN it should update the analysis jobs
+    assert updated_analysis.failed_jobs
```

Comment:
I am a bit confused why it looks for failed_jobs, but this might be me not knowing trailblazer that well.

Reply:
Yes, it is misnamed in the code. It actually contains all jobs. I will make an issue so we can fix it in a separate PR.

Reply:
yess I've was struggling with this in the past. Very misleading